serviceLibrary/src/main/java/info/mqtt/android/service/ping/AlarmPingSender.kt

```diff
@@ -53,21 +53,31 @@ internal class AlarmPingSender(
     override fun schedule(delayInMilliseconds: Long) {
         Timber.d("$id: Schedule next alarm at ${sdf.format(Date(System.currentTimeMillis() + delayInMilliseconds))}")
 
-        val pingWork = OneTimeWorkRequest.Builder(PingWorker::class.java)
-        val data = Data.Builder()
-        data.putBoolean("logging", pingLogging)
-        data.putInt("keepRecordCount", keepPingRecords)
-        data.putString("id", id)
-
-        pingWork
-            .setInitialDelay(delayInMilliseconds, TimeUnit.MILLISECONDS)
-            .setInputData(data.build())
-
-        workManager.enqueueUniqueWork(
-            "${PING_JOB}_$id",
-            ExistingWorkPolicy.REPLACE,
-            pingWork.build()
-        )
+        // Check if work is already scheduled with this ID
+        val workInfo = workManager.getWorkInfosForUniqueWork("${PING_JOB}_$id").get()
+
+        // Only schedule if no work exists or all existing work is finished/cancelled
+        if (workInfo.isEmpty() || workInfo.all { it.state.isFinished }) {
+            val pingWork = OneTimeWorkRequest.Builder(PingWorker::class.java)
+            val data = Data.Builder()
+            data.putBoolean("logging", pingLogging)
+            data.putInt("keepRecordCount", keepPingRecords)
+            data.putString("id", id)
+
+            pingWork
+                .setInitialDelay(30, TimeUnit.SECONDS)
```

Comment:
@minhdangoz May I ask, why you use 30 seconds for `setInitialDelay` instead of given millis ?

<img width="545" alt="image" src="https://github.com/user-attachments/assets/644c506f-be4b-469a-ae16-5bb64ff51b22" />

Reply:
Without an answer, I revert this
@minhdangoz